modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -233,30 +234,24 @@
                  :standard-deviation-aggregations]]
   (defmethod driver/supports? [:mongo feature] [_driver _feature] true))
 
-(defn- db-version [db]
-  (get-in db [:details :version]))
+(defmethod driver/database-supports? [:mongo :expressions]
+  [_driver _feature db]
+  (boolean (some-> (:dbms_version db)
+                   :semantic-version
+                   (driver.u/semantic-version-gte [4 2]))))
```

Comment:
Can we not do nil punning here, and require `(:dbms_version db)` to not return nil? I can imagine if `(:dbms_version db)` returns nil in the future and suddenly we're skipping tests because of it.